src/main/java/com/powsybl/balances_adjustment/pevf/PevfExchangesXml.java

```diff
@@ -0,0 +1,345 @@
+                    context.tags.put(xmlReader.getLocalName() + "." + CODING_SCHEME, codingScheme);
+                    context.tags.put(xmlReader.getLocalName(), xmlReader.getElementText());
+                    break;
+
+                case PERIOD:
+                    readPeriod(xmlReader, context);
+                    break;
+
+                case REASON:
+                    readReason(xmlReader, context);
+                    break;
+
+                default:
+                    throw new PowsyblException(UNEXPECTED_TOKEN + xmlReader.getLocalName());
+            }
+        });
+
+        // Log TimeSeries Reason
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info("TimeSeries '{}' [{}, {}] - {} ({}) : {}",  context.mRID,
+                                                                    context.period.getStart(), context.period.getEnd(),
+                                                                    context.code,
+                                                                    StandardReasonCodeType.valueOf(context.code).getDescription(),
+                                                                    context.text);
+        }
+
+        // Create DataChunk
+        DoubleDataChunk dataChunk = null;
+        // Computed number of steps
+        int nbSteps = (int) ((context.period.getEnd().toEpochMilli() - context.period.getStart().toEpochMilli()) / context.spacing.toMillis());
+        // Check if all steps are defined or not
+        if (context.positions.size() == nbSteps) {
+            // Uncompressed chunk
+            dataChunk = new UncompressedDoubleDataChunk(0, context.quantities.stream().mapToDouble(d -> d).toArray());
+        } else {
+            // Compressed chunk
+            int[] stepLengths = new int[context.positions.size()];
+            if (context.positions.size() > 1) {
+                for (int i = 1; i < context.positions.size(); i++) {
+                    int lastPosition = context.positions.get(i - 1);
+                    int newPosition = context.positions.get(i);
+                    stepLengths[i - 1] = newPosition - lastPosition;
+                }
+                // Last step is computed from nbSteps and last position value
+                stepLengths[stepLengths.length - 1] = 1 + (nbSteps - context.positions.getLast());
+            } else {
+                stepLengths[0] = nbSteps;
+            }
+            dataChunk = new CompressedDoubleDataChunk(0, nbSteps, context.quantities.stream().mapToDouble(d -> d).toArray(), stepLengths);
+        }
+
+        // Compute all instants of current time series
+        Instant[] instants = IntStream.iterate(0, i -> i + 1)
+                                      .limit(nbSteps)
+                                      .mapToObj(i -> context.period.getStart().plusMillis(i * context.spacing.toMillis()))
+                                      .toArray(Instant[]::new);
```

Comment:
I'm pretty sure it's possible to use a regular index, and check that the instant required is prior the end of the interval